src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1769,13 +1766,14 @@ function ($a) {
                                 ->children()
                                     ->arrayNode('default_middleware')
                                         ->beforeNormalization()
-                                            ->ifTrue(fn ($v) => \is_string($v) || \is_bool($v))
-                                            ->then(fn ($v) => [
-                                                'enabled' => 'allow_no_handlers' === $v ? true : $v,
+                                            ->ifString()
+                                            ->then(static fn ($v) => [
+                                                'enabled' => 'allow_no_handlers' === $v,
                                                 'allow_no_handlers' => 'allow_no_handlers' === $v,
-                                                'allow_no_senders' => true,
                                             ])
                                         ->end()
+                                        ->beforeNormalization()->ifTrue()->then(static fn ($v) => ['enabled' => true])->end()
+                                        ->beforeNormalization()->ifFalse()->then(static fn ($v) => ['enabled' => false])->end()
```

Comment:
The argument is not used.
```suggestion
                                        ->beforeNormalization()->ifTrue()->then(static fn () => ['enabled' => true])->end()
                                        ->beforeNormalization()->ifFalse()->then(static fn () => ['enabled' => false])->end()
```

But you could also add `ExprBuilder::isBool`.

```php
->beforeNormalization()->ifBool()->then(static fn ($v) => ['enabled' => $v])->end()

```

Reply:
I'd prefer not adding too much stuff in this PR (same for floats and unit-enums)